Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -1262,4 +1285,43 @@ namespace Multiplayer
         AZ::Interface<IMultiplayer>::Get()->Terminate(DisconnectReason::TerminatedByUser);
     }
     AZ_CONSOLEFREEFUNC(disconnect, AZ::ConsoleFunctorFlags::DontReplicate, "Disconnects any open multiplayer connections");
+
+    void MultiplayerSystemComponent::ExecuteDeferredCommands()
+    {
+        // If there were connect or host command invocations before the creation of the Multiplayer system
+        // then those invocations were queued. Executed them now.
+        if (const auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+        {
+            if (bool delayedHost = false;
+                settingsRegistry->Get(delayedHost, DeferredHostKey) && delayedHost)
+            {
+                settingsRegistry->Remove(DeferredHostKey);
+
+                AZ_TracePrintf("MultiplayerSystemComponent", "The Multiplayer System is now available."
+                    " Hosting now, which could not be done earlier.\n");
+                host({});
+            }
+            else if (bool delayedConnect = false;
+                settingsRegistry->Get(delayedConnect, DeferredConnectKey) && delayedConnect)
+            {
+                settingsRegistry->Remove(DeferredConnectKey);
+
+                AZ_TracePrintf("MultiplayerSystemComponent", "The Multiplayer System is now available."
+                    " Connecting now, which could not be done earlier.\n");
```

Comment:
nit: "Deferred connection now in progress"

Reply:
I dont disagree with your suggestion but I'll leave it as is for consistency, since I took this comment from deferred loadlevel example that already exists with the same log.